tools/expotools/src/check-packages/getPackagesToCheckAsync.ts

```diff
@@ -0,0 +1,57 @@
+import chalk from 'chalk';
+
+import Git from '../Git';
+import logger from '../Logger';
+import { formatCommitHash } from '../Formatter';
+import { getListOfPackagesAsync } from '../Packages';
+import { ActionOptions } from './types';
+
+const { yellow } = chalk;
+
+async function safeGetMergeBaseAsync(ref: string): Promise<string | null> {
+  try {
+    return Git.mergeBaseAsync(ref);
+  } catch (e) {
+    logger.error(`🛑 Cannot get merge base for reference: ${yellow(ref)}\n`, e.stack);
+    return null;
+  }
+}
+
+/**
+ * Resolves which packages should go through checks based on given options.
+ */
+export default async function getPackagesToCheckAsync(options: ActionOptions) {
+  const { all, packageNames } = options;
+
+  const allPackages = (await getListOfPackagesAsync()).filter((pkg) => {
+    // If the package doesn't have build or test script, just skip it.
+    return pkg.scripts.build || pkg.scripts.test;
+  });
+
+  if (all) {
+    return allPackages;
+  }
+  if (packageNames.length > 0) {
+    return allPackages.filter((pkg) => {
+      return packageNames.length === 0 || packageNames.includes(pkg.packageName);
```

Comment:
Seems clear that packageNames.length > 0 because of the check above:
```suggestion
      return packageNames.includes(pkg.packageName);
```